homeassistant/components/matter/update.py

```diff
@@ -0,0 +1,221 @@
+                | UpdateEntityFeature.PROGRESS
+                | UpdateEntityFeature.SPECIFIC_VERSION
+            )
+
+        update_state: clusters.OtaSoftwareUpdateRequestor.Enums.UpdateStateEnum = (
+            self.get_matter_attribute_value(
+                clusters.OtaSoftwareUpdateRequestor.Attributes.UpdateState
+            )
+        )
+        if (
+            update_state
+            == clusters.OtaSoftwareUpdateRequestor.Enums.UpdateStateEnum.kIdle
+        ):
+            self._attr_in_progress = False
+            return
+
+        update_progress: int = self.get_matter_attribute_value(
+            clusters.OtaSoftwareUpdateRequestor.Attributes.UpdateStateProgress
+        )
+
+        if (
+            update_state
+            == clusters.OtaSoftwareUpdateRequestor.Enums.UpdateStateEnum.kDownloading
+            and update_progress is not None
+            and update_progress > 0
+        ):
+            self._attr_in_progress = update_progress
+        else:
+            self._attr_in_progress = True
+
+    async def async_update(self) -> None:
+        """Call when the entity needs to be updated."""
+        try:
+            update_information = await self.matter_client.check_node_update(
+                node_id=self._endpoint.node.node_id
+            )
+            if not update_information:
+                self._attr_latest_version = self._attr_installed_version
+                return
+
+            self._software_update = update_information
+            self._attr_latest_version = update_information.software_version_string
+            self._attr_release_url = update_information.release_notes_url
+        except UpdateCheckError as err:
+            raise HomeAssistantError(f"Error finding applicable update: {err}") from err
+
+    async def async_added_to_hass(self) -> None:
+        """Call when the entity is added to hass."""
+        await super().async_added_to_hass()
+
+        if state := await self.async_get_last_state():
+            self._attr_latest_version = state.attributes.get(ATTR_LATEST_VERSION)
+
+        if (extra_data := await self.async_get_last_extra_data()) and (
+            matter_extra_data := MatterUpdateExtraStoredData.from_dict(
+                extra_data.as_dict()
+            )
+        ):
+            self._software_update = matter_extra_data.software_update
+        else:
+            # Check for updates when added the first time.
+            await self.async_update()
+
+    @property
+    def extra_restore_state_data(self) -> MatterUpdateExtraStoredData:
+        """Return Matter specific state data to be restored."""
+        return MatterUpdateExtraStoredData(self._software_update)
+
+    @property
+    def entity_picture(self) -> str | None:
+        """Return the entity picture to use in the frontend."""
+
+        # The Matter brand picture is not appropriate as this is the update
+        # entity for the device.
+        return None
```

Comment:
What sets the entity picture in other matter entities?

Reply:
It is not being set in other places. This is more about overwriting the default behavior of the Update entity.

Reply:
Ok, then please rewrite the docstring to make that clearer.
Maybe like this + remove the comment:
```py
    def entity_picture(self) -> str | None:
        """Return the entity picture to use in the frontend.

        This overrides UpdateEntity.entity_picture because the Matter brand picture
        is not appropriate for a matter device which has its own brand.
        """
```